### PR TITLE
Make Component friendlier to compiler

### PR DESCRIPTION
**Pull request by rsheeter**
Things I learned trying to compile glyphs: I want to see more of the fields of Component.

This is a blocker for completing the merge of https://github.com/googlefonts/fontmake-rs/pull/120.

### CODE REVIEW

**Comment by rsheeter** [comment]
Apologies for the late change but I ran a force push to rebase on a main that passes clippy

**Review by cmyr — APPROVED** [review]
Just out of curiosity, are you reading and writing these fields, or just reading? I tend to be judicious about exposing internals of things, and often prefer getters/setters as needed, but this is definitely a personal thing.

File: write-fonts/src/tables/glyf.rs [hunk 1]
```diff
@@ -514,6 +514,10 @@ impl CompositeGlyph {
             })
         }
     }
+
+    pub fn components(&self) -> impl Iterator<Item = &Component> {
+        self.components.iter()
+    }
```
[the remark this answers is not in the document]
Comment (rsheeter) [reply write-fonts/src/tables/glyf.rs:520]: Oh good call, will update.

**Comment by rsheeter** [comment]
> are you reading and writing these fields, or just reading

bbox in particular I need both. I think in general write-fonts should assume it's user might want to directly poke just about anything so providing helpful methods that do the right thing is great, but shouldn't be taken to mean we can hide direct access.

**Pull request merged**